perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -848,6 +848,34 @@ public RichMember getRichMemberWithAttributes(PerunSession sess, Member member)
 		return allowedRichMembers;
 	}
 
+	/**
+	 * Return list of RichMembers with allowed statuses contains in list of allowedStatuses and in list of allowedGroupStatuses.
+	 * If allowedGroupStatuses is empty or null, get richMembers with all statuses.
+	 *
+	 * @param sess perun session
+	 * @param richMembers rich members
+	 * @param allowedStatuses allowed statuses
+	 * @param allowedGroupStatuses allowed group statuses
+	 * @return list of allowed richMembers
+	 */
+	private List<RichMember> getOnlyRichMembersWithAllowedStatusesAndGroupStatuses(PerunSession sess, List<RichMember> richMembers, List<String> allowedStatuses, List<String> allowedGroupStatuses) {
+		List<RichMember> allowedRichMembers = new ArrayList<>();
+		if(richMembers == null || richMembers.isEmpty()) return allowedRichMembers;
+		if(allowedGroupStatuses == null || allowedGroupStatuses.isEmpty()) return richMembers;
```

Comment:
This ignores the allowedStatuses if they are set.

Reply:
Edited.